### PR TITLE
CI: Fix linting issues and enforce stricter linting in CI pipeline

### LINKED ISSUE #1413: [Security] Workflow continuous_integration.yml is using vulnerable action pre-commit/action
The workflow continuous_integration.yml is referencing action pre-commit/action using references v2.0.2. However this reference is missing the commit [80db042ff08cdddbbbfb6f89c06a6bfc4dddf0b7](https://github.com/pre-commit/action/commits/80db042ff08cdddbbbfb6f89c06a6bfc4dddf0b7) which may contain fix to the some vulnerability. 
The vulnerability fix that is missing by actions version could be related to: 
(1) CVE fix 
(2) upgrade of vulnerable dependency 
(3) fix to secret leak and others. 
Please consider to update the reference to the action.

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Fix linting issues and enforce stricter linting in CI pipeline

## Component(s) name

`arista.avd`

## Proposed changes

- Fix various linting issues (mostly newlines or spacing)
- Remove file type exclusions on file ending pre-commit hook
- Refactor pre-commit action to remove deprecated action `pre-commit/action@v2.0.2`
- In CI, run pre-commit on all files - not only the ones changed in PR.
- Remove exclusions from `ansible-test sanity` in CI and from `make sanity-lint`
- Update CI Artifacts for CVP related scenarios - not part of CI(?)
- Remove `eos_config_deploy_list` scenario. Was removed in PR #1003 but later added again by mistake in PR #1406
- Cleanup `Makefile` by removing unused commands from the file.
- Transform `make install-requirements` to manual command in CI
- Remove unused linting scripts under `.github`

## How to test

`make sanity`
`pre-commit run --all-files`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM!

File: .github/workflows/pull-request-management.yml [hunk 1]
```diff
@@ -87,7 +82,15 @@ jobs:
...
       - uses: actions/checkout@v2
-      - uses: pre-commit/action@v2.0.2
+      - name: Install requirements
+        run: |
+          pip install -r ansible_collections/arista/avd/requirements-dev.txt
+          pip install -r ansible_collections/arista/avd/requirements.txt
+      - name: Run pre-commit
+        run: |
+          git branch -a
+          pre-commit run --all-files
```
Comment (titom73) [inline .github/workflows/pull-request-management.yml]: We could use `--color always` to force output to be more readable.

File: .github/workflows/pull-request-management.yml [hunk 2]
```diff
@@ -87,7 +82,15 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
-      - uses: pre-commit/action@v2.0.2
+      - name: Install requirements
+        run: |
+          pip install -r ansible_collections/arista/avd/requirements-dev.txt
+          pip install -r ansible_collections/arista/avd/requirements.txt
+      - name: Run pre-commit
+        run: |
+          git branch -a
```
Comment (titom73) [inline .github/workflows/pull-request-management.yml]: Is this command important for the CI ? I would think it is more for debug

Comment (ClausHolbechArista) [reply .github/workflows/pull-request-management.yml]: correct, we should remove it

File: .github/workflows/pull-request-management.yml (hunk 1 above)
Comment (ClausHolbechArista) [reply .github/workflows/pull-request-management.yml]: good idea. We should add

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Review by titom73 — APPROVED** [review]
+1

**Pull request merged**